cinema/models.py

```diff
@@ -57,6 +57,10 @@ class MovieSession(models.Model):
     class Meta:
         ordering = ["-show_time"]
 
+    @property
+    def tickets_available(self) -> int:
+        return self.cinema_hall.capacity - self.tickets.count()
```

Comment:
The `tickets_available` property calculates the available tickets by subtracting the count of tickets from the cinema hall's capacity. However, this does not account for concurrency issues where multiple tickets might be booked simultaneously, potentially leading to overbooking. Consider implementing a mechanism to handle concurrent ticket bookings safely.